File: services/graphql-service/app/application/use_cases/user/get_all_users_use_case.py
from app.domain.entities.user import User
from app.application.interfaces.user_repository_interface import IUserRepository

class GetAllUsersUseCase:

    def __init__(
        self,
        repository: IUserRepository
    ):
        self.repository = repository

    async def execute(
        self
    ) -> list[User]:
        
        return await self.repository.get_all()

File: services/rabbitmq-service/consumer/src/RabbitMQService.Application/UseCases/GetAllUsers/GetAllUsersUseCase.cs
using RabbitMQService.Application.DTOs;
using RabbitMQService.Application.Interfaces;
using RabbitMQService.Application.Mappers;
using RabbitMQService.Domain.Entities;

namespace RabbitMQService.Application.UseCases.GetAllUsers;

public sealed class GetAllUsersUseCase(IUserRepository repository)
{
    public async Task<IReadOnlyList<UserDto>> ExecuteAsync(CancellationToken cancellationToken = default)
    {
        var users = await repository.GetAllAsync(cancellationToken);

        return users.Select(u => u.ToDto()).ToList();
    }
}